src/components/noteListPage/NoteListPage.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { httpMethod, httpRequest } from '../../helpers/httpRequest';
+import { INote } from '../../interfaces/note';
+import Note from '../note/Note';
+import NoteCreation from '../noteCreation/NoteCreation';
+import Search from '../search/Search';
+
+type MyProps = {};
+type MyState = { notes: INote[] };
+
+export default class NoteListPage extends Component<MyProps, MyState> {
+
+  constructor(props: {}) {
+    super(props);
+    this.state = { notes: [] };
+    this.addNote = this.addNote.bind(this);
+    this.deleteNote = this.deleteNote.bind(this);
+    this.handleMarkAsDone = this.handleMarkAsDone.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleArchive = this.handleArchive.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadNotes();
+  }
+
+  render() {
+    return (
+      <div>
+      <nav className="navbar navbar-light bg-light">
+        <span className="navbar-brand mb-0 h1">My Notes 📝</span>
+        <Link to={'/archived'}>
+            <button className="btn btn-light">archived notes</button>
+        </Link>
+        <Search handleSearch={this.handleSearch}/>
+      </nav>
+      <div className="container">
+        <NoteCreation addNote={this.addNote}/>
+        {this.state.notes.length === 0 ?
+        <div>There is no notes</div> :
+        this.state.notes.map((note) =>
+        <Note
+          note={note}
+          key={note.id}
+          deleteNote={this.deleteNote}
+          handleMarkAsDone={this.handleMarkAsDone}
+          handleArchive={this.handleArchive}
+        />)}
+      </div>
```

Comment:
Indentation needs to be more readable.

One of the alternatives:
```js
      <div className="container">
        <NoteCreation addNote={this.addNote}/>
        { this.state.notes.length === 0
            ? <div>There is no notes</div>
            : this.state.notes.map((note: INote) =>
                <Note
                  note={note}
                  key={note.id}
                  deleteNote={this.deleteNote}
                  handleMarkAsDone={this.handleMarkAsDone}
                  handleArchive={this.handleArchive}
                />,
              )
        }
      </div>
```

or:
```js
      <div className="container">
        <NoteCreation addNote={this.addNote}/>
        { this.state.notes.length === 0
            ? <div>There is no notes</div>
            : this.state.notes.map(note =>   // if you don't pass the type, you may "arrow-parens": [true, "ban-single-arg-parens"]
                <Note
                  note={note}
                  key={note.id}
                  deleteNote={this.deleteNote}
                  handleMarkAsDone={this.handleMarkAsDone}
                  handleArchive={this.handleArchive}
                />)      // if you do not like the trailing comma for some reason
        }
      </div>
```

Reply:
used your first option. Looks better!